Infrastructure change tracking (#29273)
DataDog/documentation: content/en/infrastructure/resource_catalog/resource_changes/_index.md

## Setup

To view configuration changes for your resources:

1.  **Enable Resource Collection**: Ensure that resource collection is enabled for your cloud resources so they are visible within the Resource Catalog. You can manage this setting from the [Resource Catalog settings page][2] or the relevant [cloud provider integration tile][3].
2.  **Cloud Provider Event Forwarding**: For real-time or more frequent change updates, configure event forwarding from your cloud provider:
    - **AWS**: Set up change event forwarding through AWS Config. See the [AWS Config integration page][4] for details.
    - **Azure**: Enable Resource Collection for your Azure subscriptions in the [Azure integration tile][5].
    - **Google Cloud Platform**: Follow the instructions on the [Google Cloud Platform integration page][6] to receive resource events in Datadog when Google's Cloud Asset Inventory detects changes in your cloud resources.

 <div class="alert alert-info">If you only enable Resource Collection without specific cloud provider event forwarding, Datadog uses <strong>Snapshot Changes</strong>.</div>

### Snapshot changes

> 

Reviewer: We should include a tiny note saying that Snapshot changes view can be accessed by using the _toggle_ (which may be removed in the future, fyi) in the Resource Changes page.

Author: Where is the toggle exactly? I don't see it in the Demo env. Added a generic sentence for now though, but I'd like to confirm the exact UI.

Snapshot Changes are generated by comparing crawled snapshots of your resource configurations. These snapshots are taken approximately every 5-15 minutes.

Key characteristics of Snapshot Changes:
- **No Additional Setup**: Requires no extra configuration beyond enabling resource collection.

> - **End State Reflection**: Snapshot Changes reflect the differences between snapshots. If a resource is updated and then deleted between crawls, only the deletion is reflected as the final state.

Author: I think your thought was cut off?

> - **Retention**: Datadog Snapshot Changes are retained for 2 weeks.

Reviewer: The accessibility is currently only for 1 week within the Resource Changes view

- **Cost**: Generating Snapshot Changes does not incur additional event charges.

Cloud provider changes (AWS Config, Azure Resource Graph, Google Cloud Asset Inventory) are available in Event Management with a default retention of 15 months. Forwarding these events does not incur charges for custom events from Datadog.

> 

Reviewer: Please remove this section from here. This is included under Snapshot changes. I also don't want to make commitments on pricing yet (while in Preview)

## Navigating and viewing Resource Changes

You can access Resource Changes in the Datadog platform in the following ways:

-  **Main Navigation**: Go to [**Infrastructure > Resources > Resource Changes**][8].
-  **Within a Resource's Side Panel**:
    1. Open the side panel for a specific resource (for example, by clicking on a host in the Resource Catalog or a related host view).
    2. Scroll down to the **Recent Changes** section.

> ## Querying Resource Changes

Reviewer: This is excellent. Let me get a quick review from @cborchert for an accuracy check

| First-Class Attributes  | Filter by attributes like Cloud Provider, Region, Environment, Account, etc. | `cloud_provider:aws region:us-east-1 env:staging account_id:123456789012`     |
| Resource Name           | Search for resources by their display name.                                 | `display_name:*bucket*`                                                          |
| Tags                    | Filter resources based on assigned tags.                                    | `tags:"application:frontend" "owner:team-alpha"`                                   |

### Supported search query types

- **Wildcard**: Use asterisks `*` for partial matching (before and after the term). Example: `service:*example*`
- **Union (OR)**: Use `OR` for multiple values (not available for resource types or categories). Example: `service:(service_a OR service_b)`
- **Exclusion (Empty/Not Empty)**: Use `-` prefix to exclude. Example: `-service:*` (shows resources where the service tag is not set).

## Troubleshooting with Resource Changes

> Resource Changes are invaluable during incident troubleshooting to answer the critical question: "What changed?".

Reviewer: Let's try and keep this simple. 

Can we try and simplify this to say how Resource Changes can be used similar to what's in the blog here: https://www.datadoghq.com/blog/snapshot-changes/

> When a monitor alerts, you might find an "Infrastructure Changes" link or section within the Monitor status page or related views. This directs you to the Resource Changes page, pre-filtered with context from the alert (like relevant tags, time frame, and service). This allows you to:

Reviewer: let's add "Env" here (which is a reserved tag)

- **Quickly identify relevant changes**: See a consolidated view of the latest changes for resources that could be impacting the alerting service.
- **Broaden your search**: Investigate changes beyond your immediate purview that might affect your service's reliability, especially for untagged or shared infrastructure.

> - **Drill down into change details**: Select a change to view specifics, such as the exact configuration modification (often as a side-by-side diff), and sometimes, related logs from cloud providers (like AWS CloudTrail) to understand who made the change and why.

Reviewer: The basic gist is this. Allows you to see one centralized page to inspect for infrastructure changes that happened in the same timeframe as an incident for the impacted environment, service etc. 

The motion for users is to inspect changes that seem risky and then get more details (side-by-side diff, Cloud Trail logs) etc.

> By correlating infrastructure changes with incident timelines, you can significantly reduce the Mean Time To Resolution (MTTR).

Reviewer: Ok to include at the beginning or feel free to skip. This is a conjecture.

---
title: Resource Changes
further_reading:
- link: "/infrastructure/resource_catalog/"
  tag: "Documentation"
  text: "Datadog Resource Catalog Overview"
- link: "/integrations/amazon_config/"
  tag: "Documentation"
  text: "AWS Config Integration"
- link: "/integrations/google_cloud_platform/#resource-changes-collection"
  tag: "Documentation"
  text: "Google Cloud Platform Integration for Resource Changes"
- link: "/integrations/azure/"
  tag: "Documentation"
  text: "Azure Integration for Resource Changes"
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">Resource Changes is not supported for your selected <a href="/getting_started/site">Datadog site</a> ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

{{< callout url="https://www.datadoghq.com/product-preview/recent-changes-tab/" >}}

>   esource Changes is in Preview. Click <strong>Request Access</strong> and complete the form to request access.

Reviewer 2: ```suggestion
Resource Changes is in Preview. Click <strong>Request Access</strong> and complete the form to request access.
```

- Networking
- Monitoring
- Identity
- Data Processing

For a comprehensive list of supported resources, see the [Supported resources](#supported-resources) section.

## Setup

To view configuration changes for your resources:

1.  **Enable Resource Collection**: Ensure that resource collection is enabled for your cloud resources so they are visible within the Resource Catalog. You can manage this setting from the [Resource Catalog settings page][2] or the relevant [cloud provider integration tile][3]. This step automatically enables [Snapshot Changes](#snapshot-changes) for your resources.
2.  **Cloud Provider Event Forwarding**: For real-time or more frequent change updates, configure event forwarding from your cloud provider:
    - **AWS**: Set up change event forwarding through AWS Config. See the [AWS Config integration page][4] for details.
    - **Azure**: Enable Resource Collection for your Azure subscriptions in the [Azure integration tile][5].
    - **Google Cloud Platform**: Follow the instructions on the [Google Cloud Platform integration page][6] to receive resource events in Datadog when Google's Cloud Asset Inventory detects changes in your cloud resources.

>  <div class="alert alert-info">If you only enable Resource Collection without specific cloud provider event forwarding, Datadog uses <strong>Snapshot Changes</strong>.</div>

Reviewer 2: ```suggestion
 <div class="alert alert-info">If you enable only Resource Collection without setting up cloud provider event forwarding, Datadog identifies configuration changes using <strong>Snapshot Changes</strong>.</div>
```

Reviewer 2: My suggestion to make this note a little clearer. Might not be needed based on the other comment.

### Snapshot Changes

> Snapshot Changes represent Datadog inferred changes to cloud resource configurations and relationships. They are generated by comparing snapshots of your resource configurations, taken approximately every 5-15 minutes. 

Reviewer 2: ```suggestion
Snapshot Changes represent configuration and relationship updates that Datadog detects by comparing snapshots of your cloud resources, taken approximately every 5-15 minutes.
```

To view Snapshot Changes, toggle **Snapshot Changes** on [**Infrastructure > Resources > Changes**][8].

Key characteristics of Snapshot Changes:
- **No Additional Setup**: Requires no extra configuration beyond enabling resource collection.
- **End State Reflection**: Snapshot Changes reflect the differences between snapshots. If a resource is updated and then deleted between crawls, only the deletion is reflected as the final state.
- **Retention**: Datadog Snapshot Changes are retained for one week.
- **Cost**: Generating Snapshot Changes does not incur additional event charges.

## Navigating and viewing Resource Changes

You can access Resource Changes in the Datadog platform in the following ways:

> -  **Main Navigation**: Go to [**Infrastructure > Resources > Changes**][8].

Author: Will this change before we release the docs?

> -  **Main Navigation**: Go to [**Infrastructure > Resources > Changes**][8].

Reviewer: It won't change before we release the docs, nop

### Supported search query types

- **Wildcard**: Use asterisks `*` for partial matching (before and after the term). Example: `service:*example*`
- **Union (OR)**: Use `OR` for multiple values (not available for resource types or categories). Example: `service:(service_a OR service_b)`
- **Exclusion (Empty/Not Empty)**: Use `-` prefix to exclude. Example: `-service:*` (shows resources where the service tag is not set).

## Troubleshooting with Resource Changes

During an incident, the most critical question is often "What changed?". Resource Changes helps you answer this.

When a monitor alerts, you are linked directly to the **Resource Changes** page. This page is pre-filtered with important context from the alert, such as the relevant time frame and any associated tags (like `service`, `env`, or `team`).

Use this centralized page to:

- **Identify risky changes**: Scan the list of modifications that occurred during the incident window for the impacted service, environment, and related shared infrastructure.

> - **Investigate the details**: For any suspicious change, select it to view the exact configuration modification as a side-by-side diff and related logs from cloud providers (like AWS CloudTrail) to understand who made the change and why.

Reviewer 2: ```suggestion
- **Investigate the details**: Select any suspicious change to view a side-by-side configuration diff and related logs (such as AWS CloudTrail) that help identify who made the change and why.
```
